src/Interpreters/InterpreterShowColumnsQuery.cpp

```diff
@@ -155,14 +155,19 @@ WHERE
 
     if (query.limit_length)
         rewritten_query += fmt::format(" LIMIT {}", query.limit_length);
-
     return rewritten_query;
 }
 
 
 BlockIO InterpreterShowColumnsQuery::execute()
 {
-    return executeQuery(getRewrittenQuery(), getContext(), true).second;
+    /// Disable enable_adaptive_reorder_short_circuit_arguments here, since reorder the position of
+    /// short circuit function's arguments will cause some invalid expressions to be executed.
```

Comment:
Why it can cause invalid expressions to be executed?

Reply:
I guess it comes from 
```
inner_type = 'Decimal' AND toInt8(decimal_scale_and_precision[1]) <= 65 AND toInt8(decimal_scale_and_precision[2]) <= 30,
```
If `inner_type != 'Decimal'`, the following expressions should not be exected. But if we enable `reorder`, them will be executed.

Maybe we need to set `enable_adaptive_reorder_short_circuit_arguments=false` at default, since the above example is a reasonable case.

Reply:
> Maybe we need to set enable_adaptive_reorder_short_circuit_arguments=false at default, since the above example is a reasonable case.

Yes, also use cases like this:
`number != 0 and intDiv(..., number)` - we cannot reorder arguments here